C/common/pipeline_debugger.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Fledge pipeline debugger class
+ *
+ * Copyright (c) 2025 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Mark Riddoch
+ */
+#include <pipeline_debugger.h>
+
+using namespace std;
+
+/**
+ * Constructor for the pipeline element debugger
+ */
+PipelineDebugger::PipelineDebugger() : m_buffer(NULL)
+{
+}
+
+/**
+ * Destructor for the pipeline element debugger
+ */
+PipelineDebugger::~PipelineDebugger()
+{
+	if (m_buffer)
```

Comment:
FYI: in C++, calling _delete_ on a null pointer is legal and harmless. I wouldn't change anything but it can save a lot of testing for a null pointer.